public/js/showoff.js

```diff
@@ -745,6 +745,16 @@ function currentSlideFromName(name) {
   	    found = count;
   	    return false;
   	  }
+      var dataSection = $(slide).attr("data-section").toLowerCase();
+      // firstText is usually a header for the slide
+      var firstText = $(slide).find(".content :first").text().replace(/ /g, '_').toLowerCase();
```

Comment:
What about `...text().replace(/[\W]+/g, '_').replace(/_+$/, '')...`

In otherwords, all non-word characters are underscored, or removed if it's the end of the string?

Reply:
That's what GitHub uses, except they hyphenate instead of underscore. (Example: https://github.com/binford2k/goatops/blob/master/README.md#are-you-frustrated-with-bullshit-in-your-it-job) Do you have a preference?

Reply:
I guess the github style is what most people are used to.